### PR TITLE
Refactor ml module

### PR DESCRIPTION
**Pull request by floscha**
This PR does some basic refactoring of the `koalas.ml` module. For example, it adds type annotations (as suggested in #151), moves constants to respective variables, fixes docstrings, and removes unnecessary code.

### CODE REVIEW

File: databricks/koalas/ml.py [hunk 1]
```diff
@@ -14,13 +14,24 @@
...
 
-def corr(kdf, method='pearson'):
+if TYPE_CHECKING:
+    import databricks.koalas as ks
+
+
+CORRELATION_OUTPUT_COLUMN = '_correlation_output'
+
+
+def corr(kdf: 'ks.DataFrame', method: str = 'pearson') -> pd.DataFrame:
```
Comment (rxin) [inline databricks/koalas/ml.py]: one thing that i think would really help with making this more understandable is to add an example (doctest?).

File: databricks/koalas/ml.py [hunk 2]
```diff
@@ -43,13 +54,13 @@ def corr(kdf, method='pearson'):
     return arr
 
 
-def to_numeric_df(kdf):
+def to_numeric_df(kdf: 'ks.DataFrame') -> Tuple[pyspark.sql.DataFrame, List[str]]:
     """
     Takes a dataframe and turns it into a dataframe containing a single numerical
     vector of doubles. This dataframe has a single field called '_1'.
```
Comment (rxin) [inline databricks/koalas/ml.py]: same here ... an example would be great

**Comment by rxin** [comment]
Looks great. Thanks @floscha. I left some comments, mostly about using doctest to help with readability.

**Comment by floscha** [comment]
Hi @rxin, I've added doctests for both methods and made sure they pass. Tbh, I believe the `ml` module could need some further refactoring but I'd leave those to a future PR :wink:

**Comment by rxin** [comment]
Thanks. Merging.

File: databricks/koalas/ml.py [hunk 3]
```diff
@@ -31,10 +42,15 @@ def corr(kdf, method='pearson'):
...
                    * spearman : Spearman rank correlation
     :return: :class:`pandas.DataFrame`
+
+    >>> ks.DataFrame({'A': [0, 1], 'B': [1, 0], 'C': ['x', 'y']}).corr()
+         A    B
+    A  1.0 -1.0
+    B -1.0  1.0
     """
-    assert method in ('pearson', 'spearman'), method
+    assert method in ('pearson', 'spearman')
```
Comment (thunterdb) [inline databricks/koalas/ml.py]: why?

File: databricks/koalas/ml.py [hunk 4]
```diff
@@ -14,13 +14,24 @@
...
 from pyspark.ml.feature import VectorAssembler
 from pyspark.ml.stat import Correlation
 
 
-def corr(kdf, method='pearson'):
+if TYPE_CHECKING:
+    import databricks.koalas as ks
+
+
+CORRELATION_OUTPUT_COLUMN = '_correlation_output'
```
Comment (thunterdb) [inline databricks/koalas/ml.py]: it is good style to prefix it with `_koalas` to make sure it does not clash

File: databricks/koalas/ml.py (hunk 2 above)
Comment (thunterdb) [reply databricks/koalas/ml.py]: That function is more meant to be internal

File: databricks/koalas/ml.py (hunk 3 above)
Comment (floscha) [reply databricks/koalas/ml.py]: What was the point of the `, method` at the end of the line?

File: databricks/koalas/ml.py (hunk 4 above)
Comment (floscha) [reply databricks/koalas/ml.py]: Fair enough. I might change it to `_koalas_corr` in a future pull request.

**Pull request merged**